src/App.js

```diff
@@ -1,19 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const updateChatData = (updatedChatData) => {
+    setChatData(
+      chatData.map((entry) =>
+        entry.id === updatedChatData.id ? updatedChatData : entry
+      )
+    );
+  };
+
+  const calcTotalHearts = (chatData) => {
+    return chatData.reduce((total, entry) => {
```

Comment:
👍 Nice use of `reduce` to tally up the liked message count from state we were already tracking, without the need for an additional piece of state.